src/NBitcoin/ChainBase.cs

```diff
@@ -95,11 +131,23 @@ public bool SameTip(ChainBase chain)
             return this.Tip.HashBlock == chain.Tip.HashBlock;
         }
 
+        /// <summary>
+        /// Gets the target work required for a network and a specific block height.
+        /// </summary>
+        /// <param name="network">The network to use for work computation.</param>
+        /// <param name="height">The height of the block to get work for.</param>
+        /// <returns>The target work.</returns>
         public Target GetWorkRequired(Network network, int height)
         {
             return this.GetBlock(height).GetWorkRequired(network);
         }
 
+        /// <summary>
+        /// Validates the chain on a given network.
+        /// </summary>
+        /// <param name="network">Network to validate the chain on.</param>
+        /// <param name="fullChain">Whether to validate full chain (all blocks) or just the tip.</param>
+        /// <returns>Whether the chain is valid or not.</returns>
         public bool Validate(Network network, bool fullChain = true)
```

Comment:
is this method used anywhere?
it seems like a heavy one, should not be used that much (at all?)

Reply:
No references to it. I will delete it.